libsplinter/src/network/connection_manager/mod.rs

```diff
@@ -62,7 +62,7 @@ impl ConnectionManager {
 
     pub fn start(&mut self) -> Result<Connector, ConnectionManagerError> {
         let (sender, recv) = sync_channel(CHANNEL_CAPACITY);
-        let connection_state = self.connection_state.clone();
+        let mut state = self.connection_state.clone();
```

Comment:
Is this state only used by the thread created below?  If so, it doesn't need to be a field.  It should just owned by the thread.